app/src/main/java/com/example/imageEditor/apiService/ApiImpl.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.imageEditor.apiService
+
+import com.example.imageEditor.COLLECTION_ENDPOINT
+import com.example.imageEditor.PAGE
+import com.example.imageEditor.base.OnListenProcess
+import com.example.imageEditor.getMethodHttp
+import com.example.imageEditor.model.CollectionModel
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import java.lang.reflect.Type
+import java.util.concurrent.Callable
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.Future
+
+class ApiImpl(private val onListenProcess: OnListenProcess) : Api {
+    private val executorService: ExecutorService = Executors.newCachedThreadPool()
+
+    override fun getCollections(page: Int): List<CollectionModel> {
+        var result: List<CollectionModel> = emptyList()
+        onListenProcess.onProgress()
+        val future: Future<List<CollectionModel>> =
+            executorService.submit(
+                Callable {
+                    try {
+                        val query = mapOf(Pair(PAGE, page.toString()))
+                        val listType: Type? = object : TypeToken<List<CollectionModel>>() {}.type
+                        return@Callable Gson().fromJson(
+                            getMethodHttp(COLLECTION_ENDPOINT, query),
+                            listType,
```

Comment:
```suggestion
                        val query = mapOf(Pair(PAGE, page.toString()))
                        val listType: Type? = object : TypeToken<List<CollectionModel>>() {}.type
                        return@Callable Gson().fromJson(
                            getMethodHttp(COLLECTION_ENDPOINT, query),
                            listType,
```
chỗ này hard code quá? nghiên cứu generics type để dùng parser nhiều model khác nhau em nhé!